<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<style type="text/css">
ol, ul { margin: 0.5em; }
li { margin: 0.2em; }
</style>
<title>NGS mapper ROC curves</title>
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html');">
<div id="wrap">
<div id="header"><h1>NGS mapper ROC curves</h1></div>
<span id="top"></span>
<div id="widemain">
<!-- BEGIN OF THE BODY -->
<h3>The ROC curves</h3>
<table width=100%>
<tr><td width=50%><img width=100% src="images/100se-10.png"/><td><img width=100% src="images/100pe-10.png"/></tr>
</table>
<table>
<tr><th>Program<th>Version<th>Options<th>100k 100bp SE<th>100k 2x100bp PE (CPU sec)</tr>
<tr><th>bowtie2<td>2.0.0-beta3<td>(default)<td>78.1<td>154.0</tr>
<tr><th>bwa<td>0.5.9-r26-dev<td>(default)<td>106.5<td>230.1</tr>
<tr><th>bwa-sw<td>0.5.9-r26-dev<td>(default)<td>237.4<td>502.0</tr>
<tr><th>bwa-sw64<td>0.6.0-r69-dev<td>(default)<td>139.4<td>286.5</tr>
<tr><th>gsnap<td>2011-10-16<td>(default)<td>98.9<td>538.9</tr>
<tr><th>novoalign<td>2.05.33<td>-k14 -s3<td>359.7<td>349.5</tr>
<tr><th>smalt<td>~2011-10-17<td>-k20 -s13<td>468.8<td>640.2</tr>
</table>

<h3>Simulated data</h3>
100,000 reads (read pairs) are simulated the human genome with <a
href="https://github.com/lh3/wgsim">wgsim</a>.  In this simulation, we first
simulate a diploid genome containing about 28.6 million SNPs and short INDELs
and then simulate <i>error free</i> reads from the diploid genome. Although
reads are error free, many reads cannot be perfectly mapped to the reference
genome due to the presence of variations. The exact wgsim command lines are
(the first for single-end and the second for paired-end):
</p>
<textarea name="code" class="cpp" cols="60" rows="10">
wgsim -N 100000 -r 0.01 -1 100 -S11 -d0 -e0 hs37m.fa r0.fq /dev/null
wgsim -N 100000 -r 0.01 -1 100 -2 100 -S11 -e0 hs37m.fa r1.fq r2.fq
</textarea>
<br>
<h3>Evaluated programs and command lines</h3>
Default configurations are attempted unless the default certainly fails or the
documentation explicitly suggests better configurations for 100bp HiSeq reads.
Probably the options in use are suboptimal for bowtie-v1 and soap2. The
detailed command lines are:

<textarea name="code" class="cpp" cols="60" rows="10">
bowtie -S -e 200 hs37m -q r0.fq r0.bowtie.sam
bowtie2 -x hs37m -f r0.fa > r0.bowtie.sam
bwa aln hs37m.fa r0.fa > r0.sai && bwa samse hs37m.fa r0.sai r0.fa > r0.bwa.sam
bwa bwasw hs37m.fa r0.fa > r0.bwasw.sam
gsnap -A sam -d hs37m r0.fa > r0.gsnap.sam
novoalign -r Random -o SAM -f r0.fq -d hs37m-k14s3.novo > r0.novo.sam
smalt map -f samsoft -o r0.smalt.sam hs37m-k20s13 r0.fq
soap -D hs37m.fa.index -a r0.fq -o r0.soap -g 3 -l 32
stampy.py -g hs37m -h hs37m -o r0.stampy.sam -M r0.fq
</textarea>

Paired-end reads are mapped with the following command lines:
<textarea name="code" class="cpp" cols="60" rows="10">
bowtie2 -x hs37m -q -1 r1.fq -2 r2.fq -S r12.bowtie2.sam
bwa aln hs37m.fa r1.fq > r1.sai && bwa aln hs37m.fa r2.fq > r2.sai \
    && bwa sampe hs37m r1.sai r2.sai r1.fq r2.fq > r12.bwa.sam
bwa bwasw ../index/bwa/hs37m.fa r12.fq > r12.bwasw.sam
gsnap -A sam -d hs37m r1.fq r2.fq > r12.gsnap.sam
novoalign -r Random -o SAM -f r1.fq r2.fq -i 500 50 -d hs37m-k14s3.novo > r12.novo.sam
smalt map -f samsoft -i 650 -o r12.smalt-k20s13.sam hs37m-k20s13 r1.fq r2.fq
stampy.py -g hs37m -h hs37m -o r12.stampy.sam -M r1.fq,r2.fq
soap -D hs37m.fa.index -a r1.fq -b r2.fq -l 32 -g 3 -u dummy -2 dummy -o r12.soap
</textarea>
<br>
<h3>Evaluation</h3>
The accuracy of each mapper is evaluated by the wgsim_eval.pl script which
regards a mapping is correct if it is within 50bp of the simulated position.
The command line is:
<textarea name="code" class="cpp" cols="60" rows="10">
wgsim_eval.pl alneval -ag50 xxx.sam > xxx.roc
</textarea>


<link type="text/css" rel="stylesheet" href="SyntaxHighlighter.css"></link>
<script language="javascript" src="shCore.js"></script>
<script language="javascript" src="shBrushCpp.js"></script>
<script language="javascript">
  dp.SyntaxHighlighter.HighlightAll('code');
</script>
<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2009-11-19</p>
</div>
<!-- END OF THE FOOTER -->
</div>
<!-- Piwik -->
<script type="text/javascript">
var pkBaseURL = (("https:" == document.location.protocol) ? "https://sourceforge.net/userapps/piwik/lh3lh3/" : "http://sourceforge.net/userapps/piwik/lh3lh3/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
piwik_action_name = '';
piwik_idsite = 1;
piwik_url = pkBaseURL + "piwik.php";
piwik_log(piwik_action_name, piwik_idsite, piwik_url);
</script>
<object><noscript><p><img src="http://sourceforge.net/userapps/piwik/lh3lh3/piwik.php?idsite=1" alt="piwik"/></p></noscript></object>
<!-- End Piwik Tag -->
</body>
</html>
