<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<style type="text/css">
ol, ul { margin: 0.5em; }
li { margin: 0.2em; }
</style>
<title>NGS Alignment Viewers</title>
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html');">
<div id="wrap">
<div id="header"><h1>NGS Alignment Viewers</h1></div>
<span id="top"></span>
<div id="widemain">
<!-- BEGIN OF THE BODY -->
<h2>Introduction</h2>
<p>
  Here I only focus on the software packages that are capable of viewing
  the entire 40X alignment against the human
  genome. <a href="http://staden.sourceforge.net/manual/gap4_unix_toc.html">Gap4</a>,
  <a href="http://www.phrap.org/consed/consed.html">Consed</a>,
  <a href="http://amos.sourceforge.net/hawkeye/">Hawkeye</a> and
  <a href="http://bioinformatics.bc.edu/marthlab/EagleView">EagleView</a>
  are all good assembly viewers, but it is impractical to use them on
  huge alignments/assemblies.
</p>
<br>
<h2>Software</h2>
<ul>
  <li><a href="http://bamview.sourceforge.net/">BamView</a>.
	<ul>
	  <li>Pros: Good paired-end support (similar to Gap5)
	  <li>Cons: very limited features; unable to show each base; UI can be improved.
	</ul>
  <li><a href="http://sourceforge.net/project/showfiles.php?group_id=100316&package_id=317355">Gap5</a>,
	the successor of the famous Gap4.
	<ul>
	  <li>Pros: support editing, padding and quick overview.
	  <li>Cons: difficult to install
	  [<a href="http://seqanswers.com/forums/showthread.php?t=1474">reference</a>].
	</ul>
  <li><a href="http://genomeview.sourceforge.net/">GenomeView</a>. I
	have not tried it yet and thus not shown in the table below.
  <li><a href="http://www.broad.mit.edu/igv/">Integrative Genomics
	  Viewer</a> (IGV), a versatile viewer with the support of the
	  SAM/BAM format recently.
	<ul>
	  <li>Pros: support annotation tracks; load data on the IGV server;
		fast viewing over network
	  <li>Cons: may have problem when network is not available (version
		1.3 beta)
	</ul>
  <li><a href="http://evolution.sysu.edu.cn/mapview/">MapView</a>
	[PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19369497">19369497</a>],
	a user friendly alignment viewer in active development.
	<ul>
	  <li>Pros: user friendly
	  <li>Cons: short read only; no indel; limited compression
	</ul>
  <li><a href="http://maq.sourceforge.net/maqview.shtml">Maqview</a>,
	the MAQ only alignment viewer.
	<ul>
	  <li>Pros: user friendly.
	  <li>Cons: no indel; no PE; MAQ alignment format only; windows
		version is ported by the 3rd party
	</ul>
  <li><a href="http://samtools.sourceforge.net/tview.shtml">SAMtools
	  tview</a>
	  [PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19505943">19505943</a>],
	  a small text alignment viewer.
	<ul>
	  <li>Pros: fast across networks; open alignments on FTP/HTTP.
	  <li>Cons: less fancy; limited zooming (via terminal); no PE
		support.
	</ul>
  <li><a href="http://bioinf.scri.ac.uk/tablet/">Tablet</a>.
	<ul>
	  <li>Pros: elegant and responsive interface; supporting many
		formats
	  <li>Cons: Limited features (e.g. no quality display, no PE);
		annotation display can be improved; too strict on the BAM format.
	</ul>
  <li>Others. I have
	tried <a href="http://code.google.com/p/gambit-viewer/">Gambit</a> and
	<a href="http://bioinformatics.zj.cn/magicviewer/">MagicViewer</a>,
	but they either crashed or failed to display the alignment. My
	example alignment file is a mix of short reads and long Sanger reads
	in a 5Mbp region extracted from two whole human genome
	alignments. It may contain minor errors (e.g. lack of mates), but a
	good viewer should be robust to them.
</ul>
<br>
<table>
<tr><th>Features<th>BamView<th>Gap5<th>IGV<th>MapView<th>Maqview<th>SAMtools<th>Tablet</tr>
<tr><th>Native Format<td>SAM<td>Gap5<td>SAM<td>MVF<td>MAQ<td>SAM<td>SAM+</tr>
<tr><th>User Interface<td>Java<td>X11/tk<td>Java<td>.NET<td>OpenGL<td>Curses<td>Java</tr>
<tr><th>Unix-like<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td id="tbyes">Yes</tr>
<tr><th>Windows<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">3rd-party<td bgcolor="#90FF90">Yes<td id="tbyes">Yes</tr>
<tr><th>Zoom<td bgcolor="#FFFFDD">Partial<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">Via terminal<td id="tbyes">Yes</tr>
<tr><th>Display mapQ<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">Partial<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">4 colors<td id="tbno">No</tr>
<tr><th>Gradient baseQ<td id="tbno">No<td id="tbyes">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">4 colors<td id="tbno">No</tr>
<tr><th>Paired-end<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">Partial<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FFFFDD">Orphan only<td id="tbno">No</tr>
<tr><th>Long reads<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td id="tbyes">Yes</tr>
<tr><th>Deletion<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td id="tbyes">Yes</tr>
<tr><th>Insertion<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#FFFFDD">Partial<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td id="tbno">No</tr>
<tr><th>Padding<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td id="tbno">No</tr>
<tr><th>Editing<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td id="tbno">No</tr>
<tr><th>Annotation Track<td bgcolor="#FFFFDD">Limited<td bgcolor="#FF9090">No<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FF9090">No<td bgcolor="#FFFFDD">Limited</tr>
<tr><th>Multiple tracks<td id="tbno">No<td id="tbno">No<td id="tbyes">Yes<td id="tbno">No<td bgcolor="#90FF90">Yes<td bgcolor="#FF9090">No<td id="tbno">No</tr>
<tr><th>Open remote<td id="tbno">No<td id="tbno">No<td id="tbyes">Yes<td id="tbno">No<td id="tbno">No<td id="tbyes">Yes<td id="tbno">No</tr>
</table>
<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2010-05-28</p>
</div>
<!-- END OF THE FOOTER -->
</div>
<!-- Piwik -->
<script type="text/javascript">
var pkBaseURL = (("https:" == document.location.protocol) ? "https://sourceforge.net/userapps/piwik/lh3lh3/" : "http://sourceforge.net/userapps/piwik/lh3lh3/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
piwik_action_name = '';
piwik_idsite = 1;
piwik_url = pkBaseURL + "piwik.php";
piwik_log(piwik_action_name, piwik_idsite, piwik_url);
</script>
<object><noscript><p><img src="http://sourceforge.net/userapps/piwik/lh3lh3/piwik.php?idsite=1" alt="piwik"/></p></noscript></object>
<!-- End Piwik Tag -->
</body>
</html>
