<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<title>SNPable Regions</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html');">
<div id="wrap">
<div id="header"><h1>SNPable Regions</h1></div>
<span id="top"></span>
<div id="widemain">
<!-- BEGIN OF THE BODY -->
<p>
  (Unfinished page!)
</p>
<br>
<!--
<h2>Background</h2>
<p>
  In read alignment, we always want to map read to the original genome
  because which regions should be masked is questionable and because
  masking usually leads to more false alignments. In SNP calling,
  however, we may prefer masking sometimes to achieve a uniform way to
  define which SNPs are reliable and to
</p>
<br>
-->
<h2>Procedure</h2>
<p>
  The following gives the procedure to generate the mask for single-end
  reads of length `<i>k</i>' and stringency `<i>r</i>'. Here we take
  k=35 and r=0.5. All the source codes are
  available <a href="download/seqbility-20091110.tar.bz2">here</a>,
  released under the MIT/X11 license.
<ul>
  <li>Extract all overlapping <i>k</i>-mer subsequences as read
	sequences. Command:
	<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
	  splitfa genome.fa 35 | split -l 20000000
	</textarea>
  <li>Align all reads to the genome with BWA. Other aligners would work
	if they do global alignment w.r.t. reads and give suboptimal
	hits. The preferred command-line for <tt>aln</tt> is (NOT tested,
	though):
	<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
	  bwa aln -R 1000000 -O 3 -E 3 genome.index xxaa > xxaa.sai
	</textarea>
	The default command-line would also work, but in that configuration,
	you can only get the approximate number of 1-mismatch hits rather
	than 1-difference hits; in addition, suboptimal hits would not be
	counted if there are more than 30 perfect hits (for BWA-0.4.9).
  <li>Suppose the unsorted BWA alignment results
	are <tt>xx??.sam.gz</tt>. Generate rawMask with:
	<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
	  gzip -dc xx??.sam.gz | gen_raw_mask.pl > rawMask_35.fa
	</textarea>
	When you read a sequence in rawMask_35.fa into a C
	array <tt>seq[]</tt>, you can get the approximate number of perfect
	hits (<tt>c1</tt>) and the approximate number of 1-mismatch hits
	(<tt>c2</tt>) for the <i>k</i>-mer at <tt>[x,x+k-1]</tt> with the
	following C code (NB: 63 to bypass the `&gt;' symbol which may cause
	problem in a FASTA file):
	<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
	  int c1, c2, c = seq[x] - 63;
	  c1 = c >> 3; c2 = c & 7;
	  c1 = c1? 1<<(c1-1) : 0; c2 = c2? 1<<(c2-1) : 0;
	</textarea>
	Note that <tt>c1</tt> may be zero in long `<tt>N</tt>' regions; if
	you use the default configuration of BWA-0.4.9, <tt>c2</tt> is
	always zero for <tt>c1&gt;30</tt>.
  <li>Generate the final mask:
	<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
	  gen_mask -l 35 -r 0.5 rawMask_35.fa > mask_35_50.fa
	</textarea>
	The format of this file is described in the following section.
</ul>
</p>
<br>

<h2>Using the Mask</h2>
<h3>Acquiring the mask files</h3>
<p>
  For the moment, the 35-mer rawMask file for the human genome is
  available <a href="ftp://ftp.sanger.ac.uk/pub/1000genomes/refMask">here</a>
  and the <tt>(35,0.5)</tt>
  mask <a href="ftp://ftp.sanger.ac.uk/pub/1000genomes/reference/">here</a>. You
  can easily generate mask for different <tt>r</tt> from the rawMask
  with <a href="download/seqbility-20091110.tar.bz2">my programs</a>.
</p>
<h3>File format</h3>
<p>
  File <tt>mask_35_50.fa</tt> is a fasta-like file with sequences
  composed of 0, 1, 2 and 3. Given a character <tt>c</tt> at
  position <tt>x</tt>:
  <ul>
	<li><tt>c=3</tt>: the majortiy of overlapping 35-mers are mapped
	  uniquely and without 1-mismatch (or 1-difference, depending on the
	  BWA command line) hits.
	<li><tt>c=2</tt>: the majority of overlapping 35-mers are unique and <tt>c!=3</tt>.
	<li><tt>c=1</tt>: the majority of overlapping 35-mers are non-unique.
	<li><tt>c=0</tt>: all the 35-mers overlapping <tt>x</tt> cannot be
	  mapped due to excessive ambiguous bases.
  </ul>
</p>
<h3>Applying the mask</h3>
<p>
  Given the genome file <tt>genome.fa</tt>, change all bases
  corresponding to <tt>c!=3</tt> to lowercases.
<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  apply_mask_s mask_35_50.fa genome.fa > genome.mask.fa
</textarea>
  Given a list of sites with the first two columns describing 1-based
  chromosomal positions, filter out all sites corresponding
  to <tt>c!=3</tt>:
<textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  apply_mask_l mask_35_50.fa in.list > out.list
</textarea>
</p>
<br>
<h2>SNPable Regions in the Human Genome</h2>
<p>
  35bp single-end, default BWA configuration:
</p>
<table>
  <tr><th>-r<th>2-away<th>1-away<th>HapMap2</tr>
  <tr><th>0.1<td>86.0%<td>91.6%</tr>
  <tr><th>0.2<td>84.8%<td>90.6%</tr>
  <tr><th>0.3<td>83.2%<td>89.3%</tr>
  <tr><th>0.4<td>81.9%<td>88.3%</tr>
  <tr><th>0.5<td>80.3%<td>86.8%<td>98.6%</tr>
  <tr><th>0.6<td>79.0%<td>85.6%</tr>
  <tr><th>0.7<td>77.1%<td>83.7%</tr>
  <tr><th>0.8<td>75.6%<td>82.2%<td>96.5%</tr>
  <tr><th>0.9<td>73.1%<td>79.7%</tr>
  <tr><th>1.0<td>70.7%<td>77.4%</tr>
</table>

<link type="text/css" rel="stylesheet" href="SyntaxHighlighter.css"></link>
<script language="javascript" src="shCore.js"></script>
<script language="javascript" src="shBrushCpp.js"></script>
<script language="javascript">
  dp.SyntaxHighlighter.HighlightAll('code');
</script>
<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2009-11-10</p>
</div>
<!-- END OF THE FOOTER -->
</div>
<!-- Piwik -->
<script type="text/javascript">
var pkBaseURL = (("https:" == document.location.protocol) ? "https://sourceforge.net/userapps/piwik/lh3lh3/" : "http://sourceforge.net/userapps/piwik/lh3lh3/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
piwik_action_name = '';
piwik_idsite = 1;
piwik_url = pkBaseURL + "piwik.php";
piwik_log(piwik_action_name, piwik_idsite, piwik_url);
</script>
<object><noscript><p><img src="http://sourceforge.net/userapps/piwik/lh3lh3/piwik.php?idsite=1" alt="piwik"/></p></noscript></object>
<!-- End Piwik Tag -->
</body>
</html>
