<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<style type="text/css">
ol, ul { margin: 0.5em; }
li { margin: 0.5em; }
</style>
<title>Is SW-SIMD alignment always faster?</title>
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html');">
<div id="wrap">
<div id="header"><h1>Is SW-SIMD alignment always faster?</h1></div>
<span id="top"></span>
<div id="widemain">
<!-- BEGIN OF THE BODY -->
<h2>Background</h2>
<p>Most of modern CPUs come
with <a href="http://en.wikipedia.org/wiki/SIMD">SIMD</a> instructions
(or vector instructions) which process multiple blocks of data with one
CPU instruction and thus considerably reduce the running time of
vectorized programs. This technology has been used to speed up the
standard Smith-Waterman alignment for over ten years, and a recent paper
by <a href="http://farrar.michael.googlepages.com/">Michael Farrar</a>
shows that a further 2-8X speed-up over existing SIMD-based algorithms
can be achieved by careful vectorization using
<a href="http://en.wikipedia.org/wiki/SSE2">SSE2</a> instructions. This
paper stimulates general interest on SIMD-based alignment algorithm, and
substantially influences the development of several programs, including
SSEARCH (modified by Farrar), HMMER3, NovoAlign, SHRiMP, SOAP2 and
BWT-SW. All these programs prove that SIMD-based algorithm is
faster. But my question is: is it always faster than the best non-SIMD
implementation?</p>
<p>To answer this question, I decided to do the following benchmark.</p>
<br>
<h2>Benchmark</h2>
<h3>Design</h3>
<p>First of all, the speed of a program varies with the type of input. I
do not intend to evaluate the performance on all types of inputs, but
only focus on two examples which is related to my work at hand.</p>

<p>The first data set consists of one chick protein ENSGALP00000012484
and the full set of human proteins from Ensembl v53, and the second set
consists of the 4,639,675bp <i>E. coli</i> K-12 strain (AC:NC_000913.2)
and ten 300bp simulated sequences by my wgsim program in
the <a href="http://samtools.sourceforge.net">samtools</a> package. The
command-line for the simulation is:
<ul><pre>wgsim -1 300 -2 300 -R 0.2 -r 0.05 -N 10 K12.fa test.fq /dev/null</pre></ul>

Sequences are aligned with four programs: the stdsw component
of <a href="http://maq.sourceforge.net/bwa-man.shtml">BWA-0.4.6-4</a>,
swat from <a href="http://www.phrap.org/">Phrap-080730</a>, ssearch
from <a href="http://fasta.bioch.virginia.edu/fasta_www2/fasta_list2.shtml">FASTA-35.4.7</a>
and swsse2, with the exception that swsse2 does not work for long DNA
sequences. Swsse2 requires SSE2 support. Ssearch can be optionally
compiled with SSE2 support. Both BWA and swat do not explicitly call
SSE2 instructions, although the compiler may optimize them implicitly
with SSE2.
</p>
<p>I used the BLOSUM62 matrix (gap open -10; gap extension -2) for
protein alignment and BLASTN scoring system (gap open: -5; gap extension
-2) for DNA alignment.
</p>
<h3>Results</h3>
<p>The following results were obtained on a Linux machine having two
1.86GHz L5320 Intel CPUs with SSE2 support. GCC-4.1.2 were used to
compile the three programs.</p>

<p>First, the protein alignment:</p>
<table>
  <tr><th>Program<th>CPU (s)<th>Mem (MB)<th>Command-line</tr>
  <tr><th>bwa<td>103.9<td>3.7<td>bwa stdsw -p chick.fa -T 500 hs_pep.fa</tr>
  <tr><th>ssearch<td>154.0<td>30.0<td>ssearch35 -s BL62 -f 10 -g 2 -E 1e-50 chick.fa hs_pep.fa</tr>
  <tr><th>ssearch-sse2<td>8.4<td>30.0<td>ssearch35 -s BL62 -f 10 -g 2 -E 1e-50 chick.fa hs_pep.fa</tr>
  <tr><th>swat<td>178.7<td>8.68<td>swat chick.fa hs_pep.fa -M BL62 -E 1e-20</tr>
  <tr><th>swsse2<td>5.7<td>0.7<td>swsse2 -t 500 bl62.mat chick.fa hs_pep.fa</tr>
  <tr><th>swsse2-w<td>48.8<td>0.7<td>swsse2 -w -t 500 bl62.mat chick.fa hs_pep.fa</tr>
  <tr><th>swsse2-n<td>150.6<td>0.7<td>swsse2 -n -t 500 bl62.mat chick.fa hs_pep.fa</tr>
</table>
<p>SSE2-powered aligners are an order of magnitude faster than the
non-SIMD implementations.</p>

<p>However, the DNA alignment benchmark yields quite a different
result:</p>
<table>
  <tr><th>Program<th>CPU (s)<th>Mem (MB)<th>Command-line</tr>
  <tr><th>bwa<td>141<td>28.0<td>bwa stdsw -T 50 K12.fa test.fa</tr>
  <tr><th>ssearch<td>3563<td>19.2<td>ssearch35 -r "+1/-3" -f 5 -g 2 -E 1e-50 test.fa K12.fa</tr>
  <tr><th>ssearch-sse2<td>217<td>19.2<td>ssearch35 -r "+1/-3" -f 5 -g 2 -E 1e-50 test.fa K12.fa</tr>
  <tr><th>swat<td>457<td>1497.9<td>swat test.fa K12.fa -M blastn.mat (on both strands)</tr>
</table>

<p>In this table, bwa-gcc is the fastest. Anyway, no matter what
compiler is used, bwa approaches the SSE2-optimized ssearch in speed,
and is over an order of magnitude faster than the standard ssearch in
the FASTA package. Swat achieves a similar speed to bwa.</p>

<h3>Further comments on speed</h3>
<ul>
  <li>Benchmark done previously mainly focused on two sets of short
	protein sequences, instead of a set of short DNA sequences against
	one long DNA sequence.
  <li>Ssearch outputs suboptimal hits, while bwa only records the second
	best suboptimal hit with a heuristic (inexact) rule. Apparently,
	swat does not detect the suboptimal hits given one long reference
	sequence. Detecting suboptimal hits causes overhead. Using the
	default E-value cutoff 2.0 with ssearch makes it 50% slower.
  <li>BWA learns from swat, but it improves swat by using a novel linear
	space algorithm. By the way, in another case consisting of two sets
	of short sequences, swat outperforms BWA.
  <li>According to Rognes and Seeberg, 2000, swat optimizations were
	also implemented in ssearch by that time.
  <li>The SSE2 optimization in ssearch was written by Michael Farrar and
	therefore we would expect ssearch and swsse2 (Michael's program in
	the publication) have similar performance.
  <li>According to
	the <a href="http://faculty.virginia.edu/wrpearson/fasta/changes_v34.html">ChangeLog</a>
	of FASTA v34, the SSE2-optimized ssearch is 8-10X faster than the
	standard implementation, which is consistent with my benchmark.
  <li>Rognes and Seeberg, 2000, showed that their program, swmmx, was 6X
	as fast as ssearch-3.2t08. Farrar, 2007, showed that swsse2 is 2-8X
	faster than swmmx, depending on the scoring system.
</ul>

<h3>Conclusion</h3>
<ul>
  <li>The standard ssearch seems not the best non-SIMD implementation.
  <li>At least for some types of inputs, some non-SIMD Smith-Waterman
	implementation may outperform the best SIMD implementation to date.
  <li>It is yet to known if SIMD may further accelerate swat and the SW
	component of bwa.
</ul>

<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2009-04-26</p>
</div>
<!-- END OF THE FOOTER -->
</div>
</body>
</html>
