<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<title>Software and Database</title>
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html', 'inc/sb-main.html');">
<div id="wrap">
<div id="header"><h1>Software and Database</h1></div>
<span id="top"></span>
<div id="main">
<!-- BEGIN OF THE BODY -->
<h2>Ongoing Projects</h2>
<ul>
  <li><a href="http://samtools.sourceforge.net">SAMtools</a>: SAM
	(Sequence Alignment/Map) format is a generic alignment format for
	large-scale nucleotide alignment. SAMtools is a set of utilities
	that process alignments in the SAM format. I am one of the main
	contributors to
	the <a href="http://samtools.sourceforge.net/SAM1.pdf">SAM
	specification</a> and the developer of C version of
	SAMtools.</li>
  <li><a href="http://bio-bwa.sourceforge.net">BWA</a>: BWA is a new
	read alignment program based on Burrows-Wheeler Transform (BWT). It
	short-read alignment mode is an order of magnitude faster than MAQ
	while achieving similar alignment accuracy. The long-read mode is
	several times faster than SSAHA2 and BLAT. I am the
	developer.<br></li>
  <li><a href="http://maq.sourceforge.net">MAQ</a> (Map and Assembly
	with Qualities): MAQ is a set of programs that maps short reads to
	the reference and assembles the consensus from the alignment. I am
	the chief developer.</li>
  <li><a href="http://treesoft.sourceforge.net">Treesoft</a>:
	TreeSoft provides several softwares for reconstructing or
	manipulating phylogenetic trees. I wrote NJTREE and initial codes
	for TreeFam.</li>
  <li><a href="http://www.treefam.org">TreeFam</a> (Tree families
	databases): TreeFam is a database of phylogenetic trees of animal
	genes. I am one of the key founders of this
	project.</li>
</ul>
<h2>Historical Projects</h2>
<ul>
  <li><a href="http://sourceforge.net/projects/trace-utils/">Trace
	Utilities</a>: Trace-utils is a program that recovers heterozygotes
	from trace files. This program works and can produce reasonable
	results. Some of its original features are still missing in similar
	softwares developed recently. Trace-utils was written in 2004 but
	has not been really finished.</li>
</ul>
<h2>Involved Projects</h2>
<ul>
  <li><a href="http://pig.genomics.org.cn">PigGIS</a> (Pig Genomics
	Informatics System): PigGIS is a pig genome database based on 0.66X
	WGS reads, 0.7 million ESTs and 1 million GenBank records. Jue Ruan
	implemented the whole database, Yiran Guo wrote the paper and I
	designed the strategy.</li>
  <li><a href="http://snap.humgen.au.dk">SNAP</a> (SNP Annotation
	Platform): SNAP integrates various information from several
	databases. SNAP was developed by Shengting Li, Lijia Ma and Soren
	Vang. I designed the basic strategy.</li>
  <li><a href="http://sourceforge.net/projects/xat/">CAT</a>
	(Cross-species Alignment Tool): CAT is able cDNAs to the genome of a
	different species. Intra-species cDNA-to-genome alignment of course
	works, too. XAT was written by Liang Guan. I designed the strategy,
	organized the paper and wrote a small part of the
	codes.<br></li>
</ul>
<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2009-02-27</p>
</div>
<!-- END OF THE FOOTER -->
</div>
<!-- Piwik -->
<script type="text/javascript">
var pkBaseURL = (("https:" == document.location.protocol) ? "https://sourceforge.net/userapps/piwik/lh3lh3/" : "http://sourceforge.net/userapps/piwik/lh3lh3/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
piwik_action_name = '';
piwik_idsite = 1;
piwik_url = pkBaseURL + "piwik.php";
piwik_log(piwik_action_name, piwik_idsite, piwik_url);
</script>
<object><noscript><p><img src="http://sourceforge.net/userapps/piwik/lh3lh3/piwik.php?idsite=1" alt="piwik"/></p></noscript></object>
<!-- End Piwik Tag -->
</body>
</html>
