<html>
<head>
<link rel="stylesheet" type="text/css" href="style.css">
<title>UCSC MySQL</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('top', 'inc/menu.html');">
<div id="wrap">
<div id="header"><h1>Connect UCSC MySQL with Perl</h1></div>
<span id="top"></span>
<div id="widemain">
<!-- BEGIN OF THE BODY -->
<h2>Background</h2>
<p>
  <a href="http://genome.ucsc.edu/">UCSC genome browser</a> is one of
  the three most widely used genome databases in the world. It is
  convenient and consistent in design and very efficient. UCSC data are
  usually retrieved in two ways: from
  the <a href="http://genome.ucsc.edu/cgi-bin/hgTracks">Genome
  Browser</a> where you get graphical view of a specified genomic
  region, or
  from <a href="http://genome.ucsc.edu/cgi-bin/hgTables">Tables</a>
  which facilitates data retrieval in batch. However, for a programmer
  like me, a thrid way is missing: data retrieval via APIs or
  command-line tools. I guess people
  think <a href="http://www.ensembl.org">Ensembl</a> is convenient
  mainly because it provides APIs to connect to the public Ensembl MySQL
  database.
</p>
<p>
  In fact, UCSC also has a public MySQL database
  (genomep:password@genome-mysql.cse.ucsc.edu:3306), but it does not
  provide Perl or Java APIs. Here I will show how to connect to the UCSC
  MySQL database using Perl.
</p>
<br>
<h2>Perl script for connecting UCSC MySQL</h2>
<h3>batchUCSC.pl</h3>
<p>
  The perl script <a href="download/batchUCSC.pl">batchUCSC.pl</a> is a
  general script to retrieve data in a region. You should refer to
  the <a href="http://genome.ucsc.edu/cgi-bin/hgTables">Tables</a> page
  for the description of each table. The script also predefines several
  rules to retrieve data. For example, the following command retrieves
  all the SNPs contained in the region `<tt>chr1:1-1000000</tt>':
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  echo "chr1 1 1000000" | batchUCSC.pl -p snp
  </textarea>
  which is equivalent to:
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  echo "chr1 1 1000000" | batchUCSC.pl -d hg18 -p 'snp129:::'
  echo "chr1 1 1000000" | batchUCSC.pl -d hg18 -p 'snp129:chromStart:chromEnd:*'
  </textarea>
</p>
<h3>A note on binning</h3>
<p>
  For small tables like refGene, UCSC simply creates MySQL indices on
  the chromosomal coordinate of a record. For large tables like snp129,
  UCSC cacluates a bin number for each record and only indexes
  chromosome name and the bin number without indexing chromosomal
  coordinates. Doing in this way is faster than simply relying on the
  MySQL indices and saves disk space as well. Detailed algorithm can be
  found in Kent <it>et al.</it> (2002) or from
  the <a href="http://samtools.sourceforge.net/SAM1.pdf">SAM format
  specification</a>. You can also read the source code of this script to
  see how the bin number is calculated and used in query.
</p>

<link type="text/css" rel="stylesheet" href="SyntaxHighlighter.css"></link>
<script language="javascript" src="shCore.js"></script>
<script language="javascript" src="shBrushCpp.js"></script>
<script language="javascript">
  dp.SyntaxHighlighter.HighlightAll('code');
</script>
<!-- END OF THE BODY-->
</div>
<!-- BEGIN OF THE FOOTER -->
<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2009-05-27</p>
</div>
<!-- END OF THE FOOTER -->
</div>
<!-- Piwik -->
<script type="text/javascript">
var pkBaseURL = (("https:" == document.location.protocol) ? "https://sourceforge.net/userapps/piwik/lh3lh3/" : "http://sourceforge.net/userapps/piwik/lh3lh3/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
piwik_action_name = '';
piwik_idsite = 1;
piwik_url = pkBaseURL + "piwik.php";
piwik_log(piwik_action_name, piwik_idsite, piwik_url);
</script>
<object><noscript><p><img src="http://sourceforge.net/userapps/piwik/lh3lh3/piwik.php?idsite=1" alt="piwik"/></p></noscript></object>
<!-- End Piwik Tag -->
</body>
</html>
